internal/clients/options.go

```diff
@@ -1,10 +1,157 @@
 package clients
 
-import "strings"
+import (
+	"log"
+	"math"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/terraform-provider-azapi/internal/retry"
+)
+
+var DefaultRetryableStatusCodes = []int{
+	http.StatusRequestTimeout,      // 408
+	http.StatusTooManyRequests,     // 429
+	http.StatusInternalServerError, // 500
+	http.StatusBadGateway,          // 502
+	http.StatusServiceUnavailable,  // 503
+	http.StatusGatewayTimeout,      // 504
+}
+var DefaultRetryableReadAfterCreateStatusCodes = []int{
+	http.StatusNotFound,  // 404
+	http.StatusForbidden, // 403
+}
 
 type RequestOptions struct {
 	Headers         map[string]string
 	QueryParameters map[string]string
+	RetryOptions    *policy.RetryOptions
+}
+
+// CombineRetryOptions combines multiple RequestOptions into a single policy.RetryOptions.
+func CombineRetryOptions(opts ...*policy.RetryOptions) *policy.RetryOptions {
+	if len(opts) == 0 {
+		return nil
+	}
+
+	statusCodes := make([]int, 0)
+	for _, opt := range opts {
+		if opt == nil {
+			continue
+		}
+		statusCodes = append(statusCodes, opt.StatusCodes...)
+	}
+
+	var maxRetries int32 = 0
+	for _, opt := range opts {
+		if opt == nil {
+			continue
+		}
+		if opt.MaxRetries > maxRetries {
+			maxRetries = opt.MaxRetries
+		}
+	}
+
+	var retryDelay time.Duration = 0
+	var maxRetryDelay time.Duration = 0
+	for _, opt := range opts {
+		if opt == nil {
+			continue
+		}
+		if retryDelay == 0 || opt.RetryDelay < retryDelay {
+			retryDelay = opt.RetryDelay
+		}
+		if opt.MaxRetryDelay > maxRetryDelay {
+			maxRetryDelay = opt.MaxRetryDelay
+		}
+	}
+
+	shouldRetry := func(resp *http.Response, err error) bool {
+		for _, opt := range opts {
+			if opt == nil {
+				continue
+			}
+			if opt.ShouldRetry != nil {
+				if opt.ShouldRetry(resp, err) {
+					return true
+				}
+			}
+		}
```

Comment:
We could keep the happy path more left aligned here

```suggestion
		for _, opt := range opts {
			if opt == nil || opt.ShouldRetry == nil {
				continue
			}

      if opt.ShouldRetry(resp, err) {
				return true
			}
		}
```